### PR TITLE
refactor: Use Mustache instead of Handlebars

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Uses the `mustache` dependency instead of `handlebars` to minimize dependency count.

## Release Notes

Milestone: 2.17.0
Changelog:
Removed `handlebars` dependency in favor of `mustache` for technical currency purposes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [x] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [x] These changes may need ported to the appropriate branches (list here):  
  - [x] **`next`**: [#2975](https://github.com/zowe/zowe-explorer-vscode/pull/2975)

### CODE REVIEW

**Review by anaxceron — APPROVED** [review]

File: packages/zowe-explorer-api/src/vscode/ui/utils/HTMLTemplate.ts [hunk 1]
```diff
@@ -21,10 +21,13 @@ const HTMLTemplate: string = `
...
         <meta 
             http-equiv="Content-Security-Policy" 
-            content="default-src 'none'; img-src vscode-resource: https:; script-src 'nonce-{{ nonce }}';
+            content="default-src 'none'; font-src data:; img-src data: vscode-resource: https:; script-src 'nonce-{{ nonce }}';
             style-src vscode-resource: 'unsafe-inline' http: https: data:;"
         />
         <base href="{{ uris.resource.build }}">
+        {{#uris.resource.css}}
+        <link type="text/css" rel="stylesheet" href="{{ uris.resource.css }}" />
+        {{/uris.resource.css}}
```
Comment (t1m0thyj) [inline packages/zowe-explorer-api/src/vscode/ui/utils/HTMLTemplate.ts:30]: Do we want to add support for CSS resources to webviews in ZE V2, or is this a change that we should postpone until V3?

Comment (traeok) [reply packages/zowe-explorer-api/src/vscode/ui/utils/HTMLTemplate.ts]: Sorry about that, yeah this should be in the v3 branch only. I've removed it in 09da4a1 👍

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Review by JillieBeanSim — COMMENTED** [review]
re-ran Theia test hoping just a glitch, if continues to fail may need to investigate issue against main for v2

**Pull request merged**